contracts/sentinel/PCVSentinel.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@rari-capital/solmate/src/utils/ReentrancyGuard.sol";
+import "../refs/CoreRef.sol";
+import "../pcv/IPCVDeposit.sol";
+import "./IPCVSentinel.sol";
+import "./IGuard.sol";
+
+/**
+ * @title PCV Sentinel
+ * @dev the PCV Sentinel should be granted the role Guardian
+ * @notice The PCV Sentinel is a automated extension of the Guardian.
+ * The Guardian can add Guards to the PCV Sentinel. Guards run checks
+ * and provide addresses and calldata for the Sentinel to run, if needed.
+ */
+contract PCVSentinel is IPCVSentinel, CoreRef, ReentrancyGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // The set of all guards
+    EnumerableSet.AddressSet private guards;
+
+    /**
+     * @notice Creates a PCV Sentinel with no guards
+     * @param _core the Tribe core address
+     */
+    constructor(address _core) CoreRef(_core) {
+        _setContractAdminRole(keccak256("PCV_SENTINEL_ADMIN_ROLE"));
```

Comment:
I think we should deprecate the "contract admin role" concept and use TribeRoles.